CMakeModules/FindMPI.cmake

```diff
@@ -231,7 +249,8 @@ else()
     # Allow all paths, and add an extra path if set
     set(PATHOPT )
     # Start with MPI_HOME from the environment, of given
-    set(_MPI_PREFIX_PATH $ENV{MPI_HOME})
+    #set(_MPI_PREFIX_PATH $ENV{MPI_HOME})
```

Comment:
I've added a comment. Basically some MPI environs set this when they are installed and this can then clash with the MPI environ specified via -DMPI=...